dataRetriever/factory/requestersContainer/metaRequestersContainerFactory.go

```diff
@@ -177,6 +182,17 @@ func (mrcf *metaRequestersContainerFactory) generateMetaChainHeaderRequesters()
 	return mrcf.container.Add(identifierHeader, requester)
 }
 
+func (mrcf *metaRequestersContainerFactory) generateReceiptRequesters() error {
+	// only one receipts topic
+	identifierReceipt := factory.ReceiptTopic
```

Comment:
```suggestion
	identifierReceipt := factory.ReceiptTopic + mrcf.shardCoordinator.CommunicationIdentifier(core.MetachainShardId)
```
in order to make it intrashard